content/yaml-notification/publish-release-notes.md

```diff
@@ -96,3 +96,39 @@ There are three supported options to set up release notes:
 ]
 {{< /highlight >}}
 
+### Generating release notes with git commits
+If your Codemagic workflow is triggered when creating a `Git tag`, you may want to automate the process of generating release notes based on your Git commits.
+You can use the **git log** command to generate release notes with commit massages between two Git tags. Here's how you can do it:
+
+{{< highlight yaml "style=paraiso-dark">}}
+  scripts:
+    name: Generating release notes with git commits
+    script: | 
+        git fetch --all --tags
+
+        prev_tag=$(git for-each-ref --sort=-creatordate  --format '%(objectname)' refs/tags | sed -n 2p )
+        notes=$(git log --pretty=format:"\n- %s" "$prev_tag"..HEAD)
+
+        echo "$notes" | tee release_notes.txt
+{{< /highlight >}}
+
+If you use this script localy it will generate release notes with all commits between tags e.x v1.0.0 and v2.0.0
+
+{{< highlight json "style=paraiso-dark">}}
+    v2.0.0
+    Fix bug A
+    Fix bug B
+{{< /highlight >}}
+
+However, when using Codemagic, you will also need to configure the `CM_CLONE_DEPTH` environment variable. By default, this variable is set to clone only one commit for builds triggered by tags. To capture all commits between tags, e.x. v1.0.0 and v2.0.0, you should set`CM_CLONE_DEPTH` to a value greater than the number of commits between those tags (e.g., 10 or more). This adjustment will ensure that the script fetches and generates release notes from the desired commit range.
+
+In your YAML file, set the value for the CM_CLONE_DEPTH variable under the environment variable section as shown below;
+{{< highlight yaml "style=paraiso-dark">}}
+workflows:
+  workflow-name:
+    environment:
+       vars:
+         CM_CLONE_DEPTH: 5
+{{< /highlight >}}
+
+Keep in mind that by setting CM_CLONE_DEPTH  value to a greater number might increase the time it takes to clone the repository during the build, so consider the trade-off between clone depth and build performance.
```

Comment:
Keep in mind that by setting CM_CLONE_DEPTH value to a greater number might increase the time it takes to clone the repository during the build, so consider the trade-off between clone depth and build performance.